core/src/main/scala/pureconfig/package.scala

```diff
@@ -143,9 +166,13 @@ package object pureconfig {
    */
   @throws[ConfigReaderException[_]]
   def loadConfigOrThrow[Config](path: Path)(implicit conv: ConfigConvert[Config], ct: ClassTag[Config]): Config = {
-    ConfigFactory.invalidateCaches()
-    val config = ConfigFactory.load(ConfigFactory.parseFile(path.toFile))
-    getResultOrThrow[Config](loadConfig[Config](config)(conv))
+    val errorOrConfig =
+      for {
+        _ <- invalidateCaches().right
+        rawConfig <- loadFile(path).right
+        config <- loadConfig[Config](rawConfig)(conv).right
+      } yield config
+    getResultOrThrow[Config](errorOrConfig)
```

Comment:
@melrief If you can't get resolution quickly, I'd be tempted to just revert 612b620 and pick up the thread in a separate issue focused on deduplication.